model/lollipop_keys.ts

```diff
@@ -0,0 +1,190 @@
+  assertionRef: AssertionRef,
+  assertionType: AssertionType,
+  expiredAt: Timestamp,
+  fiscalCode: FiscalCode,
+  pubKey: NonEmptyString,
+  status: t.union([
+    t.literal(PubKeyStatusEnum.VALID),
+    t.literal(PubKeyStatusEnum.REVOKED)
+  ])
+});
+export type NotPendingPopDocument = t.TypeOf<typeof NotPendingPopDocument>;
+
+export const PopDocumentBase = t.interface({
+  assertionRef: AssertionRef,
+  pubKey: NonEmptyString
+});
+export type PopDocumentBase = t.TypeOf<typeof PopDocumentBase>;
+
+// T type
+export const LolliPopPubKeys = t.union([
+  NotPendingPopDocument,
+  PendingPopDocument
+]);
+export type LolliPopPubKeys = t.TypeOf<typeof LolliPopPubKeys>;
+
+// TN type
+export const NewLolliPopPubKeys = t.intersection([LolliPopPubKeys, Ttl]);
+export type NewLolliPopPubKeys = t.TypeOf<typeof NewLolliPopPubKeys>;
+
+// TR type
+export const RetrievedLolliPopPubKeys = t.intersection([
+  LolliPopPubKeys,
+  RetrievedVersionedModelTTL
+]);
+export type RetrievedLolliPopPubKeys = t.TypeOf<
+  typeof RetrievedLolliPopPubKeys
+>;
+
+export class LolliPOPKeysModel extends CosmosdbModelVersionedTTL<
+  LolliPopPubKeys,
+  NewLolliPopPubKeys,
+  RetrievedLolliPopPubKeys,
+  typeof LOLLIPOPKEYS_MODEL_ID_FIELD
+> {
+  constructor(container: Container) {
+    super(
+      container,
+      NewLolliPopPubKeys,
+      RetrievedLolliPopPubKeys,
+      LOLLIPOPKEYS_MODEL_ID_FIELD
+    );
+  }
+
+  /*
+   * Reserve the key by creating a new document with version 0 with the ttl setted for the time needed,
+   * */
+  public create(
+    popDocument: LolliPopPubKeys,
+    option?: RequestOptions
+  ): TE.TaskEither<CosmosErrors, RetrievedLolliPopPubKeys> {
+    return pipe(
+      this.getTtlValue(popDocument),
+      // super.create never returns 409 error but a generic CosmosErrorResponse with io-functions-commons v26.8.1
+      TE.chain(ttl => super.create({ ...popDocument, ttl }, option))
+    );
+  }
+
+  /*
+   * Update the last version of the document setting the new properties and the ttl at 2 years
+   * */
+  public upsert(
+    popDocument: LolliPopPubKeys,
+    option?: RequestOptions
+  ): TE.TaskEither<CosmosErrors, RetrievedLolliPopPubKeys> {
+    return pipe(
+      this.getTtlValue(popDocument),
+      TE.chain(ttl => super.upsert({ ...popDocument, ttl }, option))
+    );
+  }
+
+  /**
+   * This method is disabled to avoid wrong use cases. Use upsert instead.
+   *
+   * @deprecated
+   * */
+  public update(
+    _: RetrievedLolliPopPubKeys
+  ): TE.TaskEither<CosmosErrors, never> {
+    return TE.left(
+      toCosmosErrorResponse(new Error("Cannot update a lollipop document"))
+    );
+  }
+
+  /**
+   * This method is disabled to avoid wrong use cases.
+   *
+   * @deprecated Use updateKeys instead.
+   * */
+  public updateTTLForAllVersions(): TE.TaskEither<CosmosErrors, never> {
+    return TE.left(
+      toCosmosErrorResponse(
+        new Error("Cannot update tll for old versions in a lollipop document")
```

Comment:
Ditto. I think the use of `is forbidden` is more indicative about the fact that we don't want it to happen, than the fact it's not possible to do it. Just semantics.